fix: check for spatialite-tools at configure time (#6159)
valhalla/valhalla: CHANGELOG.md

   * FIXED: Fix errors in cost_inline_tests [#6117](https://github.com/valhalla/valhalla/pull/6117)
   * FIXED: JSON serialization of NaN values [#6147](https://github.com/valhalla/valhalla/pull/6147)
   * BREAKING: apply `low_class_penalty` only when transitioning from a higher class road in `TruckCost::TransitionCost`(impacts truck routes) [#6143](https://github.com/valhalla/valhalla/pull/6143)

>    * FIXED: check for `spatialite-tools` (`spatialite_tool`/`spatialite`) at configure time so a missing dependency fails fast with a clear message instead of crashing while building `tz.sqlite` for tests [#2202](https://github.com/valhalla/valhalla/issues/2202)

Reviewer: nit: fewer words could be: 
```suggestion
   * FIXED: check for spatialite-tools at cmake config time if `ENABLE_DATA_TOOLS = ON` [#2202](https://github.com/valhalla/valhalla/issues/2202)
```

Reviewer: I might seem overly nitty. but honestly, wherever we can express things in fewer words, the better. AI made me aware how important it is to keep the mental load down wherever possible.
